graphbolt/src/index_select.cc

```diff
@@ -20,5 +23,33 @@ torch::Tensor IndexSelect(torch::Tensor input, torch::Tensor index) {
   return input.index({index.to(torch::kLong)});
 }
 
+std::tuple<torch::Tensor, torch::Tensor> IndexSelectCSC(
+    torch::Tensor indptr, torch::Tensor indices, torch::Tensor nodes) {
+  TORCH_CHECK(
+      indices.sizes().size() == 1, "IndexSelectCSC only supports 1d tensors");
+  if (indices.is_pinned() && utils::is_accessible_from_gpu(indptr) &&
+      utils::is_accessible_from_gpu(nodes)) {
+    GRAPHBOLT_DISPATCH_CUDA_ONLY_DEVICE(
+        c10::DeviceType::CUDA, "UVAIndexSelectCSC",
+        { return UVAIndexSelectCSCImpl(indptr, indices, nodes); });
+  } else if (
+      indices.device().type() == c10::DeviceType::CUDA &&
+      utils::is_accessible_from_gpu(indptr) &&
+      utils::is_accessible_from_gpu(nodes)) {
+    GRAPHBOLT_DISPATCH_CUDA_ONLY_DEVICE(
+        c10::DeviceType::CUDA, "nodesSelectCSC",
+        { return IndexSelectCSCImpl(indptr, indices, nodes); });
+  }
+  // For testing purposes, to compare with CPU implementation
```

Comment:
why do we need this in the code?

Reply:
We could also use the FusedCSCSamplingGraph in the test manually. It is to ensure that there is a CPU implementation through the same code path as well for testing purposes.

Reply:
Changed the comment.